File: 2021/Darius/day-01/solution.py
def part_one(n, arr):
    return sum(arr[i] < arr[i+1] for i in range(n-1))


# If you cared about efficiency, do this manually :x
def part_two(n, arr):
    return sum(sum(arr[i:i+3]) < sum(arr[i+1:i+4]) for i in range(n-3))


if __name__ == '__main__':
    with open("input.txt") as f:
        inp = [int(val) for val in f.readlines()]
        length = len(inp)

    print(part_one(length, inp))
    print(part_two(length, inp))

File: 2021/Darius/day-01/solution.cpp
#include <iostream>
#include <fstream>
#include <vector>


auto parse_input(std::string s) {
    int number;
    std::vector<uint32_t> input = {};
    std::ifstream File;
    File.open(s);
    while (File >> number)
        input.push_back(number);

    return input;
}

int part_one(int n, std::vector<uint32_t> input) {
    int total = 0;
    for (int i=1; i < n; i++)
        if (input[i-1] < input[i]) total++;

    return total;
}

int part_two(int n, std::vector<uint32_t> input) {
    int total = 0;
    int prev_sum = input[0] + input[1] + input[2];
    int cur_sum;
    for (int i=3; i < n; i++) {
        cur_sum = prev_sum - input[i - 3] + input[i];
        if (prev_sum < cur_sum) total++;
        prev_sum = cur_sum;
    }

    return total;
}

int main() {
    auto input = parse_input("input.txt");
    int input_size = input.size();
    std::cout << part_one(input_size, input) << std::endl;
    std::cout << part_two(input_size, input) << std::endl;
    return 0;
}
